From 95bb043daf68a13424bf3fc41b06cae05388a2e9 Mon Sep 17 00:00:00 2001
From: Erik Reider <erik.reider@protonmail.com>
Date: Wed, 1 Mar 2023 13:16:24 +0100
Subject: [PATCH] Set cairo filter to bilinear

---
 background-image.c | 1 +
 render.c           | 1 +
 2 files changed, 2 insertions(+)

diff --git a/background-image.c b/background-image.c
index e5060c19..7685ce27 100644
--- a/background-image.c
+++ b/background-image.c
@@ -430,6 +430,7 @@ void render_background_image(cairo_t *cairo, cairo_surface_t *image,
 		assert(0);
 		break;
 	}
+	cairo_pattern_set_filter(cairo_get_source(cairo), CAIRO_FILTER_BILINEAR);
 	cairo_paint_with_alpha(cairo, alpha);
 	cairo_restore(cairo);
 }
diff --git a/render.c b/render.c
index 655c5b19..6fe8807d 100644
--- a/render.c
+++ b/render.c
@@ -84,6 +84,7 @@ void render_frame_background(struct swaylock_surface *surface, bool commit) {
 	cairo_save(cairo);
 	cairo_set_operator(cairo, CAIRO_OPERATOR_SOURCE);
 	cairo_set_source_u32(cairo, state->args.colors.background);
+	cairo_pattern_set_filter(cairo_get_source(cairo), CAIRO_FILTER_BILINEAR);
 	cairo_paint(cairo);
 	if (surface->image && state->args.mode != BACKGROUND_MODE_SOLID_COLOR) {
 		cairo_set_operator(cairo, CAIRO_OPERATOR_OVER);
