diff --git a/flake.nix b/flake.nix
index 30cde42e..a88ef146 100644
--- a/flake.nix
+++ b/flake.nix
@@ -43,6 +43,7 @@
         overlays = [
           self.overlays.hyprland-packages
           self.overlays.wlroots-hyprland
+          self.overlays.extra-deps
           inputs.hyprland-protocols.overlays.default
         ];
       });
@@ -56,6 +57,7 @@
             hyprland-packages
             hyprland-extras
             wlroots-hyprland
+            extra-deps
             inputs.hyprland-protocols.overlays.default
           ]);
       };
diff --git a/nix/overlays.nix b/nix/overlays.nix
index 8a61fa10..a7974b1a 100644
--- a/nix/overlays.nix
+++ b/nix/overlays.nix
@@ -99,4 +99,14 @@ in {
       });
     };
   };
+
+  extra-deps = final: prev: {
+    wayland-protocols = prev.wayland-protocols.overrideAttrs (old: rec {
+      version = "1.32";
+      src = final.fetchurl {
+        url = "https://gitlab.freedesktop.org/wayland/${old.pname}/-/releases/${version}/downloads/${old.pname}-${version}.tar.xz";
+        hash = "sha256-dFl5nTQMgpa2le+FfAfd7yTFoJsJq2p097kmQNKxuhE=";
+      };
+    });
+  };
 }
