diff --git a/backend/drm/drm.c b/backend/drm/drm.c
index 5e334545..694e8b4c 100644
--- a/backend/drm/drm.c
+++ b/backend/drm/drm.c
@@ -448,7 +448,7 @@ static void drm_connector_state_init(struct wlr_drm_connector_state *state,
 		const struct wlr_output_state *base) {
 	*state = (struct wlr_drm_connector_state){
 		.base = base,
-		.modeset = base->allow_artifacts,
+      .modeset = base->allow_artifacts && (base->committed & WLR_OUTPUT_STATE_MODE),
 		.active = (base->committed & WLR_OUTPUT_STATE_ENABLED) ?
 			base->enabled : conn->output.enabled,
 	};
diff --git a/types/output/output.c b/types/output/output.c
index ec00d85b..83cc3d2b 100644
--- a/types/output/output.c
+++ b/types/output/output.c
@@ -822,6 +822,7 @@ bool wlr_output_commit_state(struct wlr_output *output,
 			wlr_output_update_mode(output, pending.mode);
 			break;
 		case WLR_OUTPUT_STATE_MODE_CUSTOM:
+          output->current_mode = NULL;
 			wlr_output_update_custom_mode(output,
 				pending.custom_mode.width,
 				pending.custom_mode.height,
